task.aspx.cs

```diff
@@ -9,9 +12,275 @@ namespace LogBook
+            {
+                DeleteTask();
+            }
+            else
+            {
+                Response.Write("<script>alert('Task does not Exist.');</script>");
+            }
+        }
+
+        /// <summary>Go Button Click</summary>
+        protected void LinkButton4_Click(object sender, EventArgs e)
+        {
+            GetTaskById();
+        }
+
+        ///<summary>
+        ///user defined function
+        ///</summary>
+        void GetTaskById()
+        {
+            try
+            {
+                SqlConnection con = new SqlConnection(strCon);
+                if (con.State == ConnectionState.Closed)
+                {
+                    con.Open();
+                }
+
+                SqlCommand cmd = new SqlCommand("SELECT * from task_detail_tbl where task_id='" + txtTaskId.Text.Trim() + "';", con);
+                SqlDataAdapter da = new SqlDataAdapter(cmd);
+                DataTable dt = new DataTable();
+                da.Fill(dt);
+
+                if (dt.Rows.Count >= 1)
+                {
+                    txtTaskId.Text = dt.Rows[0][1].ToString();
+                }
+                else
+                {
+                    Response.Write("<script>alert('Invalid Task ID');</script>");
+                }
+            }
+            catch (Exception ex)
+            {
+                Response.Write("<script>alert(''" + ex.Message + "');</script>");
+                
+            }
+        }
+
+        /// <summary>
+        /// Delete a task using an ID.
+        /// If succesful, the table will be cleared.
+        /// </summary>
+        void DeleteTask()
+        {
+            try
+            {
+                SqlConnection con = new SqlConnection(strCon);
+                if (con.State == ConnectionState.Closed)
+                {
+                    con.Open();
+                }
+
+                SqlCommand cmd = new SqlCommand("DELETE from task_detail_tbl WHERE task_id='" + txtTaskId.Text.Trim() + "'", con);
+
+                cmd.ExecuteNonQuery();
+                con.Close();
+                Response.Write("<script>alert('Task Deleted Successfully');</script>");
+                ClearForm();
+                GridView1.DataBind();
+
+            }
+            catch (Exception ex)
+            {
+                Response.Write("<script>alert(''" + ex.Message + "');</script>");
+            }
+        }
+
+        /// <summary>
+        /// Update a task using an ID.
+        /// If succesful, the table will be updated.
+        /// </summary>
+        void UpdateTask()
+        {
+            try
+            {
+                SqlConnection con = new SqlConnection(strCon);
+                if (con.State == ConnectionState.Closed)
+                {
+                    con.Open();
+                }
+
+                SqlCommand cmd = new SqlCommand("UPDATE task_detail_tbl SET task_name=@task_name WHERE task_id='" + txtTaskId.Text.Trim()+"'", con);
```

Comment:
### Surely you have to update more stuff right? 

I'm pretty sure you have to update more than one column. Always think ahead and think of what needs to be done.